### DISCUSSION TITLE
Expanding select word function

### DISCUSSION THREAD

**Discussion by ctwinovalon** [helix-editor/helix#11661] [Ideas] [answered]
In IntelliJ, you can use control-W to select the current word. But if you keep pressing it, it keeps expanding based on language scope, e.g., control-W selects the word in a string, control-W again selects the whole string, then the line, then the scope w/in braces, then the method, then the class, etc. Is there any way to replicate this in Helix? Not the key binding, the functionality.

**Comment by pascalkuthe** [comment, marked as answer]
`a-o` is similar. It expands to the parent tree-sitter node

**Reply by ctwinovalon** [reply]
Awesome! Thanks!

**Reply by David-Else** [reply]
This is such a massive feature, you may want to remap it to (what I consider) better keys:

```toml
[keys.normal]
"tab" = "expand_selection"
"S-tab" = "shrink_selection"
```
I think this is what VS Code did on smart select, but I forget now.

**Reply by rahil627** [reply]
_o. my. goodness._ i've been wanting this for a long time now!! i always thought x/X was supposed to do this, and couldn't figure out a macro for it... glad you asked! i actually even like the placement, (o)utside/(i)nside, is an important position (on qwerty). can also try it on mod-x/X.
